src/controllers/write/posts.js

```diff
@@ -94,3 +95,19 @@ Posts.restoreDiff = async (req, res) => {
 	helpers.formatApiResponse(200, res, await api.posts.restoreDiff(req, { ...req.params }));
 };
 
+Posts.deleteDiff = async (req, res) => {
+	if (!parseInt(req.params.pid, 10)) {
+		throw new Error('[[error:invalid-data]]');
+	}
+
+	const cid = await posts.getCidByPid(req.params.pid);
+	const isModerator = privileges.users.isModerator(cid, req.uid);
```

Comment:
Perhaps use `user.isPrivileged` here instead, so you can also capture admins and gmods.

Reply:
Fixed, my corrupted installation mislead me. #isModerator captures gmods and #isPrivileged includes mods of **any** category, so I end up adding just #isAdministrator. This also led to fixing some existing bugs.

Reply:
Okay, that's fine. We may want to revisit this and allow gmods access as well, but perhaps we can just make it a privilege.

Later :smile:

Reply:
Sure (It allows gmods access though).